Compilation Error for Industruino D21G

Trying to build an example sketch ([WebClient.ino](https://github.com/arduino-libraries/Ethernet/blob/master/examples/WebClient/WebClient.ino)) to test out the [Ethernet module](https://industruino.com/shop/product/ethernet-expansion-module-10) for the [Industruino D21G](https://industruino.com/shop/product/industruino-proto-d21g-8) (samd21g18a) results in compilation error. No issue with the depreciated [pubsubclient](https://github.com/knolleary/pubsubclient/tree/v2.8) lib.

I'm using macOS 15.3.1.

Platformio.ini:
```
[env:industruino_d21g]
platform = https://github.com/tomtobback/platform-industruino_d21g.git
board = industruino_d21g
framework = arduino
lib_deps =
	https://github.com/arduino-libraries/Ethernet.git#2.0.2
	https://github.com/hmueller01/pubsubclient3.git#v3.0 ; doesn't compile
	; https://github.com/knolleary/pubsubclient.git#v2.8
```

output:
```
Library Manager: PubSubClient3@3.0.0+sha.85f5caf has been installed!
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/atmelsam/industruino_d21g.html
PLATFORM: Atmel SAM (6.0.1+sha.70c7f03) > Industruino D21G
HARDWARE: SAMD21G18A 48MHz, 32KB RAM, 256KB Flash
DEBUG: Current (atmel-ice) External (atmel-ice, blackmagic, jlink)
PACKAGES: 
 - framework-arduino-samd-industruino @ 1.0.0+sha.f4de8cc 
 - framework-cmsis @ 1.40500.0 (4.5.0) 
 - framework-cmsis-atmel @ 1.2.1+sha.4c730d9 
 - toolchain-gccarmnoneeabi @ 1.70201.0 (7.2.1)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 9 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Ethernet @ 2.0.2+sha.d5672b7
|-- PubSubClient3 @ 3.0.0+sha.85f5caf
|-- SPI @ 1.0
Building in release mode
Compiling .pio/build/industruino_d21g/src/main.cpp.o
Compiling .pio/build/industruino_d21g/liba04/SPI/SPI.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/Dhcp.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/Dns.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/Ethernet.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/EthernetClient.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/EthernetServer.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/EthernetUdp.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/socket.cpp.o
Compiling .pio/build/industruino_d21g/lib3d4/Ethernet/utility/w5100.cpp.o
Compiling .pio/build/industruino_d21g/lib2f8/PubSubClient3/PubSubClient.cpp.o
Archiving .pio/build/industruino_d21g/liba04/libSPI.a
Indexing .pio/build/industruino_d21g/liba04/libSPI.a
Compiling .pio/build/industruino_d21g/FrameworkArduinoVariant/variant.cpp.o
In file included from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/char_traits.h:39:0,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/string:40,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/stdexcept:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/array:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/tuple:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/functional:54,
                 from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.h:107,
                 from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.cpp:8:
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:243:56: error: macro "min" passed 3 arguments, but takes just 2
     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
                                                        ^
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:265:56: error: macro "max" passed 3 arguments, but takes just 2
     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
                                                        ^
Compiling .pio/build/industruino_d21g/FrameworkArduino/IPAddress.cpp.o
Compiling .pio/build/industruino_d21g/FrameworkArduino/Print.cpp.o
Compiling .pio/build/industruino_d21g/FrameworkArduino/Reset.cpp.o
Compiling .pio/build/industruino_d21g/FrameworkArduino/SERCOM.cpp.o
Archiving .pio/build/industruino_d21g/lib3d4/libEthernet.a
In file included from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.h:10:0,
                 from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.cpp:8:
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:195:5: error: expected unqualified-id before 'const'
     min(const _Tp& __a, const _Tp& __b)
     ^
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:195:5: error: expected ')' before 'const'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:195:5: error: expected ')' before 'const'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:195:5: error: expected initializer before 'const'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:219:5: error: expected unqualified-id before 'const'
     max(const _Tp& __a, const _Tp& __b)
     ^
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:219:5: error: expected ')' before 'const'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:219:5: error: expected ')' before 'const'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:219:5: error: expected initializer before 'const'
In file included from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/char_traits.h:39:0,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/string:40,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/stdexcept:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/array:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/tuple:39,
                 from /Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/functional:54,
                 from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.h:107,
                 from .pio/libdeps/industruino_d21g/PubSubClient3/src/PubSubClient.cpp:8:
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:246:7: error: expected primary-expression before 'if'
       if (__comp(__b, __a))
       ^~
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:246:7: error: expected '}' before 'if'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:246:7: error: expected ';' before 'if'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:248:7: error: expected unqualified-id before 'return'
       return __a;
       ^~~~~~
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:268:7: error: expected primary-expression before 'if'
       if (__comp(__a, __b))
       ^~
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:268:7: error: expected '}' before 'if'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:268:7: error: expected ';' before 'if'
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:270:7: error: expected unqualified-id before 'return'
       return __a;
       ^~~~~~
/Users/sebastien/.platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/7.2.1/bits/stl_algobase.h:271:5: error: expected declaration before '}' token
     }
     ^
Indexing .pio/build/industruino_d21g/lib3d4/libEthernet.a
Compiling .pio/build/industruino_d21g/FrameworkArduino/Stream.cpp.o
*** [.pio/build/industruino_d21g/lib2f8/PubSubClient3/PubSubClient.cpp.o] Error 1
=============================================================== [FAILED] Took 5.23 seconds ===============================================================

 *  The terminal process "platformio 'run'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

```

--- a/src/PubSubClient.h
+++ b/src/PubSubClient.h
@@ -105,7 +105,7 @@
 // Maximum size of fixed header and variable length size header
 #define MQTT_MAX_HEADER_SIZE 5
 
-#if __has_include(<functional>)
+#if __has_include(<functional>) && !defined(NOFUNCTIONAL)
 #include <functional>
 /**
  * @brief Define the signature required by any callback function.

